apps/api/src/purchase-orders/purchase-orders.service.ts

```diff
@@ -0,0 +1,17 @@
+import {Injectable} from '@nestjs/common';
+import {PurchaseOrders} from "@prisma/client";
+import {PrismaService} from "../prisma.service";
+
+@Injectable()
+export class PurchaseOrdersService {
+  constructor(private prisma: PrismaService) {
+  }
+
+  async findAll(): Promise<PurchaseOrders[]> {
+    return (await this.prisma.purchaseOrders.findMany({include: {purchase_order_line_items: true}})).sort((a, b) => new Date(a.expected_delivery_date).getTime() - new Date(b.expected_delivery_date).getTime());
+  }
+
+  async findAllSorted(sortBy: string): Promise<PurchaseOrders[]> {
+    return (await this.prisma.purchaseOrders.findMany({include: {purchase_order_line_items: true}})).sort((a, b) => 0 - (b[sortBy] > a[sortBy] ? 1 : -1));
+  }
```

Comment:
Do we have recommended docs for how to use prisma? My guess is that there is a better way to pull sorted or otherwise manipulated data that I'm just not sure about.

Also, the function structure didn't seem to allow me to add any error handlers. In the future I would like to spend more time in NestJS to understand how to throw errors, etc. 

Any comments or feedback here would be appreciated.

Reply:
Under this we could actually change it to something like this.
```suggestion
  }
      return (await this.prisma.purchaseOrders.findMany({orderBy: [
	    {
	      [orderBy]: 'desc',
	    },
	    {
	      vendor_name: 'asc',
	    },
	  ],
	  include: {purchase_order_line_items: true}
	  }))
```